src/patches.js

```diff
@@ -128,14 +128,26 @@ export function applyPatches(draft, patches) {
                     break
                 case "remove":
                     if (Array.isArray(base)) {
-                        if (key === base.length - 1) base.length -= 1
-                        else
-                            throw new Error(
-                                `Remove can only remove the last key of an array, index: ${key}, length: ${
-                                    base.length
-                                }`
-                            )
-                    } else delete base[key]
+                        const index = Number(key)
+                        if (!Number.isNaN(index)) {
+                            if (index < 0 || index > base.length) {
+                                throw new Error(
+                                    "Invalid array patch: The given index is out of bounds"
+                                )
+                            }
+                            if (index === 0) {
+                                base.shift()
+                                break
```

Comment:
NIT: `if else` ... might make this code block slighly clearer? Extracting utility code might also be nice

Reply:
Probably, I'm just not a fan of a lonesome `break`.

You want `break`s removed and `else`s added? Or keep the `breaks`?

Reply:
Oh I suppose you mean to break this out into its own function?